R/method-correlation.R

```
@@ -0,0 +1,173 @@
+    ancestorIdColumns2 <- data2@ancestorIdColumns
+    allIdColumns2 <- c(recordIdColumn2, ancestorIdColumns2)
+
+
+    ## Initialize and check inputs
+    method <- veupathUtils::matchArg(method)
+    verbose <- veupathUtils::matchArg(verbose)
+
+    # Check that incoming abundance data meets requirements
+    if (!'data.table' %in% class(df1)) {
+      data.table::setDT(df1)
+    }
+
+    # Check that incoming metadata meets requirements
+    if (!'data.table' %in% class(df2)) {
+      data.table::setDT(df2)
+    }
+
+    # Check that variables is the correct class
+    if (!is.null(variables) && !'VariableMetadataList' %in% class(variables)) {
+      stop("The variables argument must be a VariableMetadataList if defined.")
+    }
+
+    computeMessage <- ''
+    veupathUtils::logWithTime(paste("Received df table with", nrow(df1), "samples and", (ncol(df1)-1), "taxa."), verbose)
+
+    # Replace NA values with 0
+    if (naToZero) {
+      veupathUtils::setNaToZero(df1)
+      veupathUtils::logWithTime("Replaced NAs with 0.", verbose)
+    }
+
+    # Ensure the sample IDs, and their order, are the same in both datasets
+    if (!identical(df1[[recordIdColumn1]], df2[[recordIdColumn2]])) {
+      if (!identical(union(df1[[recordIdColumn1]], df2[[recordIdColumn2]]), intersect(df1[[recordIdColumn1]], df2[[recordIdColumn2]]))) {
+        stop("Sample IDs do not match between data1 and data2.")
+      }
+      # Reorder df2 (sample metadata) so that it matches the order in df
+      sampleMetadata <- df2[match(df1[[recordIdColumn1]], df2[[recordIdColumn2]]), ]
+      veupathUtils::logWithTime("Reordered sampleMetadata rows based on abundance data sample id order.", verbose)
+    }
+
+    ## Extract appropriate metadata columns
+    if (!is.null(variables)) {
+      # Use the data shape to find appropriate columns in the metadata
+      inputMetadataCols <- veupathUtils::findColNamesByPredicate(variables, function(x) {identical(x@dataShape@value, "CONTINUOUS")})
```

this silently ignores requested columns? maybe the user doesnt realize categoricals dont make sense here.. they wont get any feedback unless all cols are removed.